plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -5268,6 +5268,22 @@ public void testReadFromVersionedTableWithExpiredHistory()
         assertQueryFails("SELECT * FROM " + tableName + " FOR TIMESTAMP AS OF " + timestampLiteral(v1EpochMillis, 9), "No version history table .* at or before .*");
     }
 
+    @Test
+    public void testDeleteRetainsTableHistory()
+    {
+        String tableName = "test_delete_retains_table_history_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + "(c1 INT, c2 INT)");
+        assertUpdate("INSERT INTO " + tableName + " VALUES (1, 1), (2, 2), (3, 3)", 3);
+        assertUpdate("INSERT INTO " + tableName + " VALUES (3, 3), (4, 4), (5, 5)", 3);
+        List<Long> snapshots = getTableHistory(tableName);
+
+        assertUpdate("DELETE FROM " + tableName + " WHERE c1 < 4", 4);
+        List<Long> snapshotsAfterDelete = getTableHistory(tableName);
+        assertEquals(snapshotsAfterDelete.size(), snapshots.size() + 1);
```

Comment:
I'd drop this, relax, or add a comment.

In fact we accept situation where deletion results in two snapshots, right?

Reply:
Sure. Switched it to a greater than check